### ISSUE TITLE
JS API should be able to parallelize newTable requests

### ISSUE THREAD

**Issue by niloc132** [deephaven/deephaven-core#2529]
As a developer, I want to be able to call newTable() and be able to deal with the handle or CTS before the table has finished being created. There are use cases that do not rely on having the full table finished, but only need to have the ticket so that the next request can be made.

Two API examples that could have lower latency with this change:
 * Adding a new row to an input table - as soon as the new table ticket exists, it could be sent in parallel with the DoPut to the server, so that it can be added right away to the input table
 * Fetching a table from a partitioned table - likewise, as soon as the new ticket exists which will soon correspond to a table with a single row in it, the client can ask the server to fetch a table that matches the contents of the to-be-created table
